src/charm.py

```diff
@@ -39,7 +44,7 @@ def _on_start(self, _):
         self.unit.status = ActiveStatus()
```

Comment:
you shouldn't do this unconditionally in `__init__`.
Suppose you receive website_relation_joined and you go in Blocked, next event you see (whichever event!) you'll go back to active and the user will never know that there's something wrong with the charm.

Reply:
Yes, I understand. Is there a good example of this appropriate to a (currently) simple charm like this one?

Reply:
you can look at our most recent charm: https://github.com/canonical/mimir-coordinator-k8s-operator/blob/main/src/charm.py#L103

in a nutshell: observe the collect_unit_status event, and add several 'is this healthy?' endpoints that each adds a piece of status to the event. Either blocked or active.
Ops will take the worst status and set it to the unit, on each event.